src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java

```diff
@@ -443,15 +416,14 @@ public void execute()
         return modulepathElements;
     }
 
-    private String getExecutable()
-        throws MojoFailureException
+    private JLinkExecutor getExecutor() throws MojoFailureException
     {
-        String jLinkExec;
+        JLinkExecutor jLinkExec;
         try
         {
-            jLinkExec = getJLinkExecutable();
+            jLinkExec = getJlinkExecutor();
```

Comment:
try block should be larger and not split the declaration from initialization

Reply:
This is also something not in the scope of the PR. You are asking me to do a full refactoring here, while I only want to introduce ToolProvider. This line you marked is exactly as the existing code. Please file a JIRA issue and lets create a new PR for this.